lib/utils/platform/iot_clock_afr.c

```diff
@@ -0,0 +1,215 @@
+    ullTickCount = ( uint64_t ) ( xCurrentTime.xOverflowCount ) << ( sizeof( TickType_t ) * 8 );
+
+    /* Add the current tick count. */
+    ullTickCount += xCurrentTime.xTimeOnEntering;
+
+    /* Return the ticks converted to Milliseconds */
+    return ( int64_t ) ullTickCount * _MILLISECONDS_PER_TICK;
+}
+
+/*-----------------------------------------------------------*/
+
+bool IotClock_TimerCreate( IotTimer_t * const pNewTimer,
+                           IotThreadRoutine_t expirationRoutine,
+                           void * pArgument )
+{
+    _timerInfo_t * pxTimer = ( _timerInfo_t*) pNewTimer;
+
+    IotLogDebug( "Creating new timer %p.", pNewTimer );
+
+    /* Set the timer expiration routine, argument and period */
+    pxTimer->threadRoutine = expirationRoutine;
+    pxTimer->pArgument = pArgument;
+    pxTimer->xTimerPeriod = 0;
+
+    /* Create a new FreeRTOS timer. This call will not fail because the
+    * memory for it has already been allocated, so the output parameter is
+    * also set. */
+    pxTimer->timer  = ( timer_t ) xTimerCreateStatic(  "timer",           /* Timer name. */
+                                                portMAX_DELAY,            /* Initial timer period. Timers are created disarmed. */
+                                                pdFALSE,                  /* Don't auto-reload timer. */
+                                                ( void * ) pxTimer,       /* Timer id. */
+                                                prvTimerCallback,         /* Timer expiration callback. */
+                                                &pxTimer->xTimerBuffer ); /* Pre-allocated memory for timer. */
+
+    return true;
+}
+
+/*-----------------------------------------------------------*/
+
+void IotClock_TimerDestroy( IotTimer_t * const pTimer )
+{
+    _timerInfo_t * pTimerInfo = ( _timerInfo_t * ) pTimer;
+
+    /* The value of the timer ID, set in timer_create, should not be NULL. */
+    configASSERT( pTimerInfo != NULL );
+    configASSERT( pTimerInfo->timer != NULL );
+
+    IotLogDebug( "Destroying timer %p.", pTimer );
+
+    /* Destroy the timer. */
+    if( xTimerIsTimerActive( pTimerInfo->timer ) == pdTRUE )
+    {
+        /* Stop the FreeRTOS timer. Because the timer is statically allocated, no call
+        * to xTimerDelete is necessary. The timer is stopped so that it's not referenced
+        * anywhere. xTimerStop will not fail when it has unlimited block time. */
+        ( void ) xTimerStop( pTimerInfo->timer, portMAX_DELAY );
+
+
+        /* Wait until the timer stop command is processed. */
+        while( xTimerIsTimerActive( pTimerInfo->timer ) == pdTRUE )
+        {
+            vTaskDelay( 100 );
+        }
+    }
+}
+ 
+/*-----------------------------------------------------------*/
+
+bool IotClock_TimerArm( IotTimer_t * const pTimer,
+                           uint64_t relativeTimeoutMs,
+                           uint64_t periodMs )
+{
```

Comment:
Did we agree to change this to 
```
bool IotClock_TimerArm( IotTimer_t * const pTimer,
                           uint32_t relativeTimeoutMs,
                           uint32_t periodMs )
```
?

Reply:
Yes indeed. I tried to do the PR without changing the posix impl and without changing any API, to cover those in the review, but it is time now so I updated this in both implementations.